train-core.gemspec

```diff
@@ -0,0 +1,34 @@
+# encoding: utf-8
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'train/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = 'train-core'
+  spec.version       = Train::VERSION
+  spec.authors       = ['Dominik Richter']
+  spec.email         = ['dominik.richter@gmail.com']
+  spec.summary       = 'Transport interface to talk to a selected set of backends.'
+  spec.description   = 'A minimal Train with a selected set of backends, ssh, winrm, and docker.'
+  spec.homepage      = 'https://github.com/chef/train/'
+  spec.license       = 'Apache-2.0'
+
+  spec.files = %w{train-core.gemspec README.md LICENSE Gemfile CHANGELOG.md} + Dir
+               .glob('{lib,test}/**/*', File::FNM_DOTMATCH)
+               .reject { |f| File.directory?(f) || f =~ /aws|azure|gcp/ }
```

Comment:
I think for the train-core gem we want it to only have the `local` transport. So we will want to reject all the transports except for that one. You can probably nix all the spec/tests as well.